### ISSUE TITLE
[Regression] CarouselView with Loop="False" containing ScrollView does not resize to reveal newly visible items

### ISSUE THREAD

**Issue by mfeingol** [dotnet/maui#13188]
### Description

This one's a little obscure, but is really annoying and was very hard to reproduce simply.

In short, the Label below will not become visible when the CheckBox is checked.

However, set Loop="True" or remove the ScrollView, and it will work as expected.

```xaml
        <CarouselView Loop="False" ItemsSource="{Binding Items}">
            <CarouselView.ItemTemplate>
                <DataTemplate x:DataType="local:MainPage">
                    <ScrollView>
                        <StackLayout>
                            <CheckBox IsChecked="{Binding IsChecked, Mode=TwoWay}" />
                            <Label Text="Visible when checked" IsVisible="{Binding IsChecked}" />
                        </StackLayout>
                    </ScrollView>
                </DataTemplate>
            </CarouselView.ItemTemplate>
        </CarouselView>
```

### Steps to Reproduce

Run the repro.

### Link to public reproduction project repository

https://github.com/mfeingol/repros/tree/master/CheckboxBindingRepro

### Version with bug

7.0 (current)

### Last version that worked well

7.0 (current)

### Affected platforms

Android

### Affected platform versions

11 and 13

### Did you find any workaround?

No.

**Comment by Zhanglirong-Winnie** [comment]
Verified this issue with Visual Studio Enterprise 17.7.0 Preview 2.0. Can repro on android platform with sample project.
[repros-master.zip](https://github.com/dotnet/maui/files/11879669/repros-master.zip)
![13188](https://github.com/dotnet/maui/assets/77434865/5be8586d-4070-4af4-8818-2300b77106df)

**Comment by mfeingol** [comment]
This _appears_ to be fixed with the current .NET 8 preview.

**Comment by mfeingol** [comment]
This is verified to be fixed in 8.0.61.

**Comment by PureWeen** [comment]
Thank you for following up @mfeingol
